WireAnalytics/Sources/WireAnalytics/Events/Predefined/AnalyticsEvent.UI.swift

```diff
@@ -0,0 +1,64 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+public extension AnalyticsEvent {
+
+    enum UI {
+
+        /// An event tracking when the user opens the self profile.
+
+        public static func openSelfProfile(isMigrationDotActive: Bool) -> AnalyticsEvent {
+            AnalyticsEvent(name: "ui.clicked-profile") {
+                if isMigrationDotActive {
```

Comment:
Not sure if it makes any difference, however, in the Android project they seem to also only provide the segmentation for the `true` case. (instead of `false` they would just not send the segmentation entry)

Reply:
I guess it makes sense to check. I assume all platforms should be alined here.

Reply:
On second thought, I see this is a common pattern so I expect it has already been considered